### PR TITLE
3526 - ET - Added subscription  connect/disconnect hooks

### PR DESCRIPTION
**Pull request by etelmon**
Jira Task: https://zimitio.atlassian.net/browse/ZWEB-3526

enhancement to the publication options for allowing callback functions to be executed when a subscription is connected to a subscribe or disconnected.

**Steps to Reproduce**

Run:
`npm test`

**Performance Considerations**
No performance impact.

**Sections of the Application Affected**
Look up application pr to QA.

### CODE REVIEW

**Comment by etelmon** [comment]
Do not merge yet, until I provide unit tests. It is a proof of concept.

File: lib/subscription.js [hunk 1]
```diff
@@ -437,8 +437,8 @@ function Subscription(user, subscriptionId, publication, params) {
             }
         }
         delete thisSub.publication.subscriptions[subscriptionId];
-        if (onReleaseCallback) {
-            onReleaseCallback(thisSub.tenantId);
+        if (_.isFunction(onReleaseCallback)) {
+            onReleaseCallback.call(thisSub, thisSub.tenantId, this.user, this.params);
```
Comment (Semen-Lohunov) [inline lib/subscription.js:441]: `user` and `params` seem to be redundant

Comment (etelmon) [reply lib/subscription.js]: I removed them, no longer used.

**Review by Semen-Lohunov — APPROVED** [review]
Tests passed.

**Pull request merged**